tools/nnyeah/nnyeah/AssemblyComparator/ComparingVisitor.cs

```diff
@@ -0,0 +1,144 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.MaciOS.Nnyeah;
+using Mono.Cecil;
+
+#nullable enable
+
+namespace Microsoft.MaciOS.AssemblyComparator {
+	public class ComparingVisitor {
+		ModuleDefinition EarlierModule, LaterModule;
+		bool PublicOnly;
+
+		public ComparingVisitor (ModuleDefinition earlierModule, ModuleDefinition laterModule, bool publicOnly)
+		{
+			EarlierModule = earlierModule;
+			LaterModule = laterModule;
+			PublicOnly = publicOnly;
+		}
+
+		public void Visit ()
+		{
+			var earlierElements = ModuleElements.Import (EarlierModule, PublicOnly);
+			if (earlierElements is null)
+				throw new Exception (Errors.E0007);
```

Comment:
Yes? Part of it is that this code initially didn't have the luxury on instance variables as it was all static functions. I don't see that it makes much difference since beyond the events and ctor, there is precisely one public api, `Visit`. Everything else is implementation details.